ingot-macros/src/choice.rs

```diff
@@ -273,9 +267,10 @@ pub fn attr_impl(attr: TokenStream, item: syn::ItemEnum) -> TokenStream {
         where #( #next_layer_wheres ),*
         {
             type Denom = T;
+            type Hint = #on;
 
             #[inline]
-            fn next_layer(&self) -> ::core::option::Option<Self::Denom> {
+            fn next_layer_choice(&self, hint: Option<Self::Hint>) -> ::core::option::Option<Self::Denom> {
```

Comment:
Same macro hygiene nit as PR 3/? -- `hint: ::core::option::Option<Self::Hint>` (and other such repetitions).

Reply:
Yup, fixed!